katdal/ms_extra.py

```diff
@@ -32,34 +28,39 @@
 from copy import deepcopy
 
 import numpy as np
-
-# Look for a casacore library binding that will provide Table tools
-try:
-    # Try to use the casapy table tool first
-    import casac
-    tb = casac.homefinder.find_home_by_name('tableHome').create()
-    casacore_binding = 'casapy'
-except ImportError:
-    try:
-        # Otherwise fall back to python-casacore aka pyrap
-        from casacore import tables
-        casacore_binding = 'pyrap'
-    except ImportError:
-        casacore_binding = ''
-    else:
-        # Perform python-casacore version checks
-        from pkg_resources import parse_version
-        import casacore
-
-        pyc_ver = parse_version(casacore.__version__)
-        req_ver = parse_version("2.2.1")
-
-        if not pyc_ver >= req_ver:
-            raise ImportError("python-casacore %s is required, "
-                              "but the current version is %s. "
-                              "Note that python-casacore %s "
-                              "requires at least casacore 2.3.0."
-                              % (req_ver, pyc_ver, req_ver))
+from pkg_resources import parse_version
+import casacore
+from casacore import tables
+
+# Perform python-casacore version checks
+pyc_ver = parse_version(casacore.__version__)
+req_ver = parse_version("2.2.1")
+if not pyc_ver >= req_ver:
+    raise ImportError("python-casacore %s is required, but the current version is %s. "
+                      "Note that python-casacore %s requires at least casacore 2.3.0."
+                      % (req_ver, pyc_ver, req_ver))
+
+
+def open_table(filename, readonly=False, ack=False, **kwargs):
+    """Open casacore Table."""
+    t = tables.table(filename, readonly=readonly, ack=ack, **kwargs)
+    return t if type(t) == tables.table else None
```

Comment:
Could you explain why the type check is required? Why would the constructor return anything except a tables.table?

Reply:
😅 
I think that is a wart left over from supporting casapy. In CASA-land you have the `tb` tool, and `tb.open` returns True on successfully opening the table. I then returned either the `tb` tool itself as a proxy for the `table` object, or `None` otherwise. But as you correctly point out, this does not make sense in casacore-land. In retrospect I should have done
```Python
try:
    return tables.table(...)
except RunTimeError:
    return None
```
Now that casapy is off the table, rather let the exception be raised on error.

I've now reworked both `open_table` and `open_main` into one function with no `None`s in sight.